lib/src/main/cpp/README.md

```diff
@@ -1,3 +1,129 @@
 # Traditional keypoint detection algorithms
 
-TODO
+This directory contains the native implementation of the SGBM-based depth estimation algorithm.
+
+## Description
+
+[SIFT  (Scale-Invariant Feature Transform)](https://docs.opencv.org/3.4/d7/d60/classcv_1_1SIFT.html)
+
+[SURF (Speeded-Up Robust Features)](https://docs.opencv.org/3.4/d5/df7/classcv_1_1xfeatures2d_1_1SURF.html)
+
+[ORB (Oriented FAST and Rotated BRIEF)](https://docs.opencv.org/3.4/d7/d60/classcv_1_1SIFT.html)
+
+These algorithms from OpenCV is used to
+calculate keypoints and descriptors from image. The result is:
+Firsly, vector of structures which contains floating-point numbers representing each keypoint of picture
+Secondly, a vector of descriptors accounting each keypoint which is also a vector of a floating-point numbers
+
+clang opencv contrib llvm aim path by cc cxx or set clang as highest priority compiller
+
+
+## Project structure
+
+The project consists of the following directories:
+
+- `examples` – usage examples (see below)
+- `external` - 3rd-party libraries
+- `include` – public headers to include when importing this CMake project
+- `src` – source code
+
+## Build
+
+**Note**: when building on Windows, you may be required to specify a non-default generator, for
+example, `ninja`, with `-DCMAKE_GENERATOR=<generator name>` in the configuration command, because
+Windows' default generator `nmake` seems to fail.
+
+This project requires [OpenCV](https://github.com/opencv/opencv) 4.6.0 and [OpenCV-contrib](https://github.com/opencv/opencv_contrib) contains extra modules, you should install same version as main libary. You can either
```

Comment:
Replace "you should install same version as main libary" with "of the same version"